src/averageAges.js

```diff
@@ -15,11 +15,17 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const isMan = (person) => person.sex === 'm';
+  const isManBornThisCentury = (person) =>
+    person.sex === 'm' && Math.ceil(person.died / 100) === century;
+
+  const filteredMen = people.filter(century ? isManBornThisCentury : isMan);
+
+  const averageAgeMen = filteredMen
+    .reduce((sumOfAge, person) =>
+      sumOfAge + (person.died - person.born) / filteredMen.length, 0);
+
+  return averageAgeMen;
```

Comment:
```suggestion
  const averageAgeMen = filteredMen.reduce(
    (sumOfAge, person) => sumOfAge + (person.died - person.born), 0
  );
```
А поділити на кількість можеш в самому кінці